sw/device/tests/dif_gpio/mock_mmio.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mock_mmio.h"
+
+void MmioMockTest::SetUp() {
+  if (mock_ == nullptr) {
+    // Using new because constructor is private
+    mock_.reset(new MmioMock());
+  }
+}
+
+void MmioMockTest::TearDown() {
+  if (mock_ != nullptr) {
+    // Destroy the mock object here to verify its expectations. Otherwise,
+    // it is destroyed at program exit, i.e. we end up leaking it, and its
+    // expectations are not verified.
+    mock_.reset();
+  }
+}
```

Comment:
I would not expect this to do what you want it to do. Having non-static members mutate a static member that is not guarded by a mutex makes the type globally thread-unsafe, which I'm not sure is even compatible with gtest.

Is there a reason this absolutely needs to be a static member?

Reply:
To be able to reach it from mmio functions that are mocked (Since I don't _inject_ it in the first argument of `reg32_read/write()`). For thread safety, we could use `thread_local`.